### PR TITLE
Add Fedora 40

### PR DESCRIPTION
**Pull request by osteffenrh**
# Description

Add images based on Fedora 40.

It includes:
- gcc 14.1.1 (x86, arm, aarch64, riscv, loongarch64)
- nasm 2.16.01
- Python 3.12.2, virtualenv is set up.
- Qemu 8.2 (x86, arm, aarch64, loongarch64)

Issue #94

### CODE REVIEW

**Comment by osteffenrh** [comment]
EDK2 CI Test Run: https://github.com/tianocore/edk2/pull/5646

**Comment by osteffenrh** [comment]
ping @bcran, @makubacki, @cfernald ?

**Review by cfernald — APPROVED** [review]

File: .github/workflows/Fedora-40.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# GitHub Action Workflow for building the Fedora 39 images.
```
Comment (makubacki) [inline .github/workflows/Fedora-40.yaml:1]:
```suggestion
# GitHub Action Workflow for building the Fedora 40 images.
```

**Review by makubacki — APPROVED** [review]

File: Fedora-40/fedora40_dev_entrypoint.sh [hunk 2]
```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+#
+# Copyright (c) 2023 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: BSD-2-Clause-Patent
+
+set -e
+
+#####################################################################
+# Run as the same uid/gid as the developer.
+
```
[the remark this answers is not in the document]
Comment (osteffenrh) [reply Fedora-40/fedora40_dev_entrypoint.sh:10]: This comment is in the same spot in all entrypoint scripts.
Not sure where it actually should go. I'll remove it. It does not help much anyway.

**Pull request merged**